server/src/main/java/chat/engine/Connect.java

```diff
@@ -108,13 +114,27 @@ private void closeConnection() throws IOException {
         reader.close();
         writer.close();
         clientsOnline.remove(name);
+    }
+
+    private void sendMessage(String message, String name) throws IOException {
 
+        for (Map.Entry<String, Socket> entry : clientsOnline.entrySet()) {
+            if (entry.getValue() != socket) {
+                writer = getWriter(entry.getValue());
+                writer.println(name + ": " + message);
+            }
+        }
     }
 
-    private void sendMessage(String message) throws IOException {
+    private boolean privateMessage(String message, String name) throws IOException {
+        String namePM = "";
         for (Map.Entry<String, Socket> entry : clientsOnline.entrySet()) {
-            writer = getWriter(entry.getValue());
-            writer.println(message);
+            if (message.contains(PM + (namePM = entry.getKey()))) {
+                writer = getWriter(clientsOnline.get(namePM));
+                writer.println(name + ": " + message);
```

Comment:
how user who's received PM will understand that he received exactly private message, not general. Mark this message in any way